agrc_county_project.py

```diff
@@ -154,107 +40,124 @@ def createCsvRow(row):
-            # County Project ZIP4
-            countyRow.append(0)
-            # County Project CONGRESSIONALCODE
-            countyRow.append('UT0' + str(addrRow[addrFields.index('DISTRICT')]))
-            # County Project COUNTYCODE
-            countyRow.append('UT' + addrRow[addrFields.index('CountyID')][-3:])
-            # County Project FILLER
-            countyRow.append('')
-            # County Project KEY
-            countyRow.append(addrRow[addrFields.index('OID@')])
-            # County Project LATITUDE
-            countyRow.append(addrRow[addrFields.index('SHAPE@Y')])
-            # County Project LONGITUDE
-            countyRow.append(addrRow[addrFields.index('SHAPE@X')])
-            # CountyProject row insert
-            countyProjectCursor.insertRow(countyRow)
-            # County project rows
-            counties[countyRow[0]].append(countyRow)
+    #: District identity datasets
+    congressional_districts_fc_path = Path(sgid_connection, 'SGID.POLITICAL.USCongressDistricts2012')
+    identify_result_fc_path = Path(output_folder, output_gdb_name, 'Addresses_Districts' + unique_run_id)
 
-    headerString = ','.join(countyProjectFields)
-    # List will be created for each id in county_ids
-    # Output file should be a csv
-    for county in counties:
-        countyProjectCsv = os.path.join(output_folder, '{}_{}.csv'.format(county, uniqueRunNum))
-        row_list = counties[county]
-        if len(row_list) > 0:
-            with open(countyProjectCsv, 'w') as outputFile:
-                outputFile.write(headerString + '\n')
-                for row in row_list:
-                    try:
-                        rowString = createCsvRow([str(x) for x in row])
-                        outputFile.write(rowString + '\n')
-                    except:
-                        print(row)
+    #: Make layer of only our specified counties
+    county_selection_where = None
+    if county_ids:
+        county_ids_string = ','.join(county_ids)
+        county_selection_where = f"CountyID IN ('{county_ids_string}')"
+    address_layer = 'addrpoints_' + unique_run_id
+    arcpy.MakeFeatureLayer_management(str(address_source_fc_path), address_layer, county_selection_where)
 
-                outputFile.write('\n')
+    #: Get the congressional district for each address point
+    print('Running Congressional District identify...')
+    arcpy.env.outputCoordinateSystem = arcpy.SpatialReference(4326)
+    arcpy.env.geographicTransformations = 'NAD_1983_To_WGS_1984_5'
+    arcpy.Identity_analysis(address_layer, str(congressional_districts_fc_path), str(identify_result_fc_path))
+
+    #: Copy address points w/district info to dict of counties containing list of dicts of addresses
+    #: {county: [{address, city, etc}, ...], ...}
+    print('Loading and formatting source data...')
+    counties = defaultdict(list)  #: defaultdict adds new key with an empty list if key is not yet present
```

Comment:
**thought**: i think this is the same as using `counties.setdefault('key', []).append(item)`. no need to replace it just an fyi

Reply:
That's my understanding as well. Some searching says that a defaultdict is faster than setdefault, but it probably doesn't matter here. I really just wanted an excused to try it :)